src/main/java/randoop/reflection/OperationExtractor.java

```diff
@@ -102,34 +165,44 @@ public void visit(Constructor<?> constructor) {
     if (!predicate.test(constructor)) {
       return;
     }
-    TypedClassOperation operation = TypedOperation.forConstructor(constructor);
-    addOperation(operation);
+    TypedClassOperation operation = instantiateTypes(TypedOperation.forConstructor(constructor));
+    checkSubTypes(operation);
+    if (!omitPredicate.shouldOmit(operation)) {
+      addOperation(operation);
+    }
   }
 
   /**
    * Creates a {@link MethodCall} object for the {@link Method}.
    *
+   * <p>The created operation has the declaring class of {@code method} as the declaring type. An
+   * exception is a static method for which the declaring class is not public.
+   *
    * @param method a {@link Method} object to be represented as an {@link Operation}
    */
   @Override
   public void visit(Method method) {
     if (!predicate.test(method)) {
       return;
     }
-    TypedClassOperation operation = TypedOperation.forMethod(method);
-    if (classType.isSubtypeOf(operation.getDeclaringType()) && operation.isStatic()) {
+    TypedClassOperation operation = instantiateTypes(TypedOperation.forMethod(method));
+    checkSubTypes(operation);
+
+    if (operation.isStatic()) {
+      // If this classType inherits this static method, but declaring class is not public, then
```

Comment:
The JLS says static methods may be inherited https://docs.oracle.com/javase/specs/jls/se8/html/jls-8.html#jls-8.4.8